airbyte-integrations/connectors/destination-jdbc/src/main/java/io/airbyte/integrations/destination/jdbc/copy/CopyDestination.java

```
@@ -0,0 +1,74 @@
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+package io.airbyte.integrations.destination.jdbc.copy;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import io.airbyte.db.jdbc.JdbcDatabase;
+import io.airbyte.integrations.base.Destination;
+import io.airbyte.integrations.destination.ExtendedNameTransformer;
+import io.airbyte.integrations.destination.jdbc.AbstractJdbcDestination;
+import io.airbyte.integrations.destination.jdbc.SqlOperations;
+import io.airbyte.protocol.models.AirbyteConnectionStatus;
+import io.airbyte.protocol.models.ConnectorSpecification;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public abstract class CopyDestination implements Destination {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(CopyDestination.class);
+
+  public abstract void attemptWriteToPersistence(JsonNode config) throws Exception;
+
+  public abstract ExtendedNameTransformer getNameTransformer();
+
+  public abstract JdbcDatabase getDatabase(JsonNode config) throws Exception;
+
+  public abstract SqlOperations getSqlOperations();
+
+  @Override
+  public ConnectorSpecification spec() throws Exception {
+    return AbstractJdbcDestination.getSpec();
+  }
+
+  @Override
+  public AirbyteConnectionStatus check(JsonNode config) {
+    try {
+      attemptWriteToPersistence(config);
+
+      var nameTransformer = getNameTransformer();
+      var outputSchema = nameTransformer.convertStreamName(config.get("schema").asText());
+      JdbcDatabase database = getDatabase(config);
+      AbstractJdbcDestination.attemptSQLCreateAndDropTableOperations(outputSchema, database, nameTransformer, getSqlOperations());
+
+      return new AirbyteConnectionStatus().withStatus(AirbyteConnectionStatus.Status.SUCCEEDED);
+    } catch (Exception e) {
+      LOGGER.debug("Exception while checking connection: ", e);
```

could this be `LOGGER.error`? since we swallow debug logs this ends up being hidden from the user when it would be helpful in fallback scenarios when `e.getMessage()` isn't helpful (e.g. in the case of an NPE) .